### ISSUE TITLE
Allow users create bootable ISO's even if it bigger than 4 gigabytes

### ISSUE THREAD

**Issue by W1nKille3r** [MaranBr/Systemback#8]
I encounter very annoying thing, where image is too big for squashfs. But we have a method, to ignore than limit, and create bigger iso's whatever the sblive weights.

I tried to do it myself, but failed miserably, but in short: You can "slice" squashfs to sequences, like
- filesystem.sqashfs       4194304 KB
- filesystem.sqashfs1     4194304 KB
- filesystem.squashfs2   4194304 KB
- filesystem.squashfs3   2971938 KB

The problem is: This process have to be set manually every time you make .sblive from Systemback. So its pretty much unproductive to push updates of your own image. I tried to optimize my image to be less than 4 gigs, but even on "cutted corners" it still weights like 4.62 gb at final size

Link to the solution from guy, what i found the way to fix this:
1. https://answers.launchpad.net/systemback/+question/273061
2. https://daviwa.blogspot.com/2018/05/systemback-how-to-create-systemback.html

**Comment by MaranBr** [comment]
This is a limitation of the file system. But I believe this was fixed a long time ago. It's in the changelog on the wiki page.

When converting to ISO, Systemback splits the squash.fs file into multiple files. After conversion, you can use Ventoy to boot the system directly from the ISO image.

I have ISO images larger than 6GB and everything works fine.

**Comment by W1nKille3r** [comment]
> This is a limitation of the file system. But I believe this was fixed a long time ago. It's in the changelog on the wiki page.
> 
> When converting to ISO, Systemback splits the squash.fs file into multiple files. After conversion, you can use Ventoy to boot the system directly from the ISO image.
> 
> I have ISO images larger than 6GB and everything works fine.

Hmm.. I have installed latest version (what is 1.8.703). And i checked changelog, and it has to support large ISO's. But when i trying to convert created sblive image to ISO. Button just grayed out _(even if created live images tab shows "sblive+iso")._ What is strange, but okay.

After compiling system to sblive, it automatically created ISO file with the sblive one. When im opening both ISO's in Ark, they have "same" file, because they are named the same. In UltraISO it shows that squashfs is actually splitted. And right now i tested both images where one, was auto-compiled to ISO from Systemback, and other compiled through xorriso. Seems like both booting. But i had a little quirk when trying to load something larger than 8 gigabyte squashfs.

I'm guessing, that is because squashfs have limit to 4GB per file. Grub just doesn't know what to boot first, because there 2 or more squashfs files that have same capacity. And if you have file "as you typed 6GB" it does not reach the limit of 8GB, so you have only 2 squashfs files, not 3.

I tried to put all stuff into LiveCD so i don't have to compromise with software. And seems like reaching more than 8 gigs (in my case it was compiled to 9.8GB breaks loading sequence. Im pretty sure its easy to replicate if you install as example Steam with some games that weights alot. Then try to compile LiveCD _(you may be right when saying why i would not just buy external ssd then for those tasks?)_. Im poor, and im working with things that i have on my hand so. Improvising never ending :)

**Comment by MaranBr** [comment]
I have generated much larger images, up to 9.5 GB. Everything works fine, as long as you use software that supports this type of ISO image, such as Ventoy for example. If you try to mount the image directly on the USB stick, the file system will not allow it.

**Comment by W1nKille3r** [comment]
> I have generated much larger images, up to 9.5 GB. Everything works fine, as long as you use software that supports this type of ISO image, such as Ventoy for example. If you try to mount the image directly on the USB stick, the file system will not allow it.

Anyway, after some tests, I came to the conclusion that still need to buy a portable drive for “serious” tasks, because to put everything on a flash drive, even if you can. It won't work as expected.

I spent all night compiling a 12.2 gigabyte image (ventoy shows it weighs 2.2). After loading, half of the applications refuse to start, although the main system shell (i.e. Plasma, does not crash and behaves adequately). In order for the image to be created properly. You need to have a few conditions

1. The hard disk on which the LiveCD image will be saved must have enough space, and wish it was much more, because then Systemback will show an error, and do this operation will have to start from scratch
2. Even if you want to insert a lot of software, thus making the image extremely large. Keep in mind that not many computers will be able to run it. For example, if the image weighs 10+ gigabytes, a computer with 4 gigabytes of RAM will simply not be able to operate the system properly.
3. Current version of Systemback 1.8.703, if it sees that the created image weighs more than 4 gigabytes, it automatically converts the image to ISO. So it is advisable not to delete the ISO image after conversion, otherwise you will have to convert it manually,

In short, I got a number of unknown bugs that somehow miraculously solved, before that I could not solve the problem with the limit of 4 gigabytes, and here systemback trolls me cruelly
